src/libostree/ostree-repo-commit.c

```diff
@@ -1319,6 +1484,10 @@ write_metadata_object (OstreeRepo         *self,
        */
       if (have_obj)
         {
+          if (ostree_repo_get_mode (self) == OSTREE_REPO_MODE_ARCHIVE_Z2 && self->generate_sizes)
```

Comment:
From https://github.com/ostreedev/ostree/blob/master/src/libostree/ostree-core.h#L189 it looks like the ARCHIVE_Z2 mode is just a legacy alias for the ARCHIVE mode. It may be worth changing it here to use the new value, and perhaps factor out this check into a small function, as it's repeated a few times in the patch.

Reply:
Pushed a fixup to do this